src/coreclr/jit/lower.cpp

```diff
@@ -3587,20 +3587,26 @@ GenTree* Lowering::LowerSelect(GenTreeConditional* select)
     // we could do this on x86. We currently disable if-conversion for TYP_LONG
     // on 32-bit architectures because of this.
     GenCondition selectCond;
+    GenTreeOpCC* newSelect = nullptr;
     if (((select->gtFlags & GTF_SET_FLAGS) == 0) && TryLowerConditionToFlagsNode(select, cond, &selectCond))
     {
         select->SetOper(GT_SELECTCC);
-        GenTreeOpCC* newSelect = select->AsOpCC();
+        newSelect              = select->AsOpCC();
         newSelect->gtCondition = selectCond;
         ContainCheckSelect(newSelect);
         JITDUMP("Converted to SELECTCC:\n");
         DISPTREERANGE(BlockRange(), newSelect);
         JITDUMP("\n");
-        return newSelect->gtNext;
+    }
+    else
+    {
+        ContainCheckSelect(select);
     }
 
-    ContainCheckSelect(select);
-    return select->gtNext;
+#ifdef TARGET_ARM64
+    TryLowerCselToCinc(select, cond);
```

Comment:
What happens to the Asm diff results if you only call this for the `SELECTCC` cases? How many opportunities do we miss?

Not having a `CINC` node type would simplify the code.

A follow on patch will add cinv, and if it follows this logic, then it will add both `CINV` and `CINVCC`. Just want to make sure we're not introducing a unnecessary complication.

Reply:
Disabling the `GT_CINC` related lowering did not make any difference in asmdiffs unfortunately.
I'm not sure whether we should keep only the `GT_CINCCC` part.

@jakobbotsch, any suggestions?

Reply:
I think having `GT_CINCCC` is fine given that most of the code is shared anyway.

Reply:
Okies, I'll leave `GT_CINC` and `GT_CINCC` both as is in the patch.